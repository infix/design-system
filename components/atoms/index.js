import Button from './Button'
import Input from './Input'
import Heading from './Heading'
import Text from './Text'
import IconButton from './IconButton'
import { RadioGroup, RadioButton } from './Radio'
import { Select } from './Select'

import { TwitterShareButton, FacebookShareButton } from './ShareButton'

const atoms = {
  Text,
  Heading,
  Button,
  IconButton,
  Input,
  RadioGroup,
  RadioButton,
  Select,
  TwitterShareButton,
  FacebookShareButton
}

export default atoms
