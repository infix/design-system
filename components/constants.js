/* global module */
const defaultWeights = [
  400,
  700
]

const defaultBreakpoints = [
  40,
  52,
  64
]

const defaultSpace = [
  0,
  8,
  16,
  32,
  64
]

const defaultFontSizes = [
  12,
  14,
  16,
  20,
  24,
  32,
  48,
  64,
  72
]

module.exports = {
  defaultBreakpoints,
  defaultSpace,
  defaultFontSizes,
  defaultWeights
}
