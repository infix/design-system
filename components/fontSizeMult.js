/* global require */
import {
  style,
  px
} from 'styled-system'

const {
  defaultFontSizes
} = require('./constants')

export const fontSizeMult = (propName, k) => {
  /* eslint-disable no-console */
  console.error('usage of fontSizeMult will be deprecated')
  /* eslint-enable no-console */

  return style({
    prop: 'fontSize',
    cssProperty: propName,
    key: 'fontSizes',
    transformValue: n => px(n * k),
    scale: defaultFontSizes
  })
}

export default fontSizeMult
