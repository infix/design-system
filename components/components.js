import atoms from './atoms'
import molecules from './molecules'
import organisms from './organisms'

import BarChart from './charts/BarChart'
import PieChart from './charts/PieChart'

/* OONI produced atoms */

export const Button = atoms.Button
export const Input = atoms.Input
export const RadioButton = atoms.RadioButton
export const RadioGroup = atoms.RadioGroup
export const TwitterShareButton = atoms.TwitterShareButton
export const FacebookShareButton = atoms.FacebookShareButton
export const Heading = atoms.Heading
export const Text = atoms.Text
export const Select = atoms.Select

/* OONI produced molecules */
export const InputWithIconButton = molecules.InputWithIconButton
export const LogoOONIRun = molecules.LogoOONIRun
export const Card = molecules.Card

/* OONI produced organisms */
export const Hero = organisms.Hero
export const HeroLead = organisms.HeroLead
export const Modal = organisms.Modal

/* Here we just export rebass until we implement all of these */
import {
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
} from 'rebass'

export { default as BarChart } from './charts/BarChart'
export { default as PieChart } from './charts/PieChart'

const components = {
  Button,
  Heading,
  Text,
  Input,
  Select,
  Card,
  RadioButton,
  RadioGroup,
  InputWithIconButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart,
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
}

export default components
