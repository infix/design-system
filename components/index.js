import {
  Provider,
  Flex,
  Box
} from 'rebass'

import colors from './theme/colors'
import theme from './theme/rebassTheme'

export { default as colors } from './theme/colors'
export { default as theme } from './theme/rebassTheme'

import components from './components'

// We export these from rebass directly
export {
  Provider,
  Flex,
  Box
} from 'rebass'

export const {
  Button,
  ButtonOutline,
  ButtonCircle,
  ButtonTransparent,
  Link,
  NavLink,
  BlockLink,
  Heading,
  Subhead,
  Text,
  Small,
  Lead,
  Pre,
  Code,
  Samp,
  Blockquote,
  Measure,
  Truncate,
  Label,
  Input,
  Select,
  Textarea,
  Checkbox,
  Radio,
  Slider,
  Image,
  Avatar,
  BackgroundImage,
  Container,
  Divider,
  Border,
  Media,
  Card,
  Banner,
  Panel,
  PanelHeader,
  PanelFooter,
  Progress,
  Message,
  Group,
  Toolbar,
  Badge,
  Circle,
  Tabs,
  TabItem,
  DotButton,
  Close,
  Relative,
  Absolute,
  Fixed,
  Sticky,
  Drawer,
  Overlay,
  Carousel,
  ScrollCarousel,
  CarouselSlide,
  Tooltip,
  Switch,
  Arrow,
  Star,
  Embed,
  Donut,
  Row,
  Column,
  RadioGroup,
  RadioButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  InputWithIconButton,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart
} = components

export const OONIComponents = Object.assign({}, components, { Provider, Box, Flex, colors, theme })

export default OONIComponents
