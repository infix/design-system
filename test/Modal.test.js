/* global describe, test, expect, jest */
import React from 'react'
import { renderJSON, shallowWithTheme } from './index'
import { Modal } from '../components'

describe('Modal', () => {
  test('renders', () => {
    const json = renderJSON(
      <Modal />
    )
    expect(json).toMatchSnapshot()
  })

  test('renders with close button', () => {
    const json = renderJSON(
      <Modal closeButton='right' />
    )
    expect(json).toMatchSnapshot()
  })

  test('close button click handler is called', () => {
    const onCloseClick = jest.fn()
    const wrapper = shallowWithTheme(
      <Modal show={true} closeButton='right' onHideClick={onCloseClick} />
    )
    wrapper.find('Modal__StyledCloseButton').simulate('click')
    wrapper.find('Modal__StyledModalBackdrop').simulate('click')
    expect(onCloseClick).toHaveBeenCalledTimes(2)
  })

})
