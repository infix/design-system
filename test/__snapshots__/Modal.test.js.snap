// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders 1`] = `
.c1 {
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  max-width: 100vw;
  max-height: 100vh;
  overflow: auto;
  z-index: 1050;
  background-color: #f8f9fa;
  border-radius: 2px;
}

.c0 {
  position: relative;
  display: none;
}

<Modal
  onHideClick={[Function]}
>
  <Modal__StyledModal>
    <div
      className="c0"
    >
      <Modal__StyledModalContent
        onHideClick={[Function]}
      >
        <div
          className="c1"
        />
      </Modal__StyledModalContent>
    </div>
  </Modal__StyledModal>
</Modal>
`;

exports[`Modal renders with close button 1`] = `
.c5 {
  border-radius: 56px;
  display: inline-block;
  height: 70px;
  line-height: 1;
  padding-left: 56px;
  padding-right: 56px;
  text-transform: uppercase;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  outline: 0;
  font-family: inherit;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #f8f9fa;
  background-color: #0588cb;
  border: none;
  text-align: center;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
  z-index: 1;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  cursor: pointer;
}

.c5:hover {
  border: none;
  background-color: #37a6ed;
}

.c5:active {
  -webkit-transition: .2s ease-in;
  transition: .2s ease-in;
  border: none;
  background-color: #0f77b8;
}

.c4 {
  font-size: 28px;
}

.c3 {
  font-size: 28px;
  border-radius: none;
  color: #adb5bd;
  background-color: transparent;
}

.c3:hover {
  background-color: transparent;
  color: #ced4da;
}

.c3:active {
  background-color: transparent;
  color: #868e96;
}

.c1 {
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  max-width: 100vw;
  max-height: 100vh;
  overflow: auto;
  z-index: 1050;
  background-color: #f8f9fa;
  border-radius: 2px;
}

.c0 {
  position: relative;
  display: none;
}

.c2 {
  position: absolute;
  top: 0;
  right: 0;
  height: 28px;
  padding: 0;
  margin: 0;
}

<Modal
  closeButton="right"
  onHideClick={[Function]}
>
  <Modal__StyledModal>
    <div
      className="c0"
    >
      <Modal__StyledModalContent
        closeButton="right"
        onHideClick={[Function]}
      >
        <div
          className="c1"
        >
          <Modal__StyledCloseButton
            icon={<MdClose />}
            onClick={[Function]}
            position="right"
          >
            <IconButton
              className="c2"
              fontSize={3}
              icon={<MdClose />}
              onClick={[Function]}
              position="right"
            >
              <IconButton__StyledIconButton
                className="c2"
                fontSize={3}
                icon={<MdClose />}
                onClick={[Function]}
                position="right"
              >
                <Button
                  border={2}
                  className="c2 c3"
                  fontSize={3}
                  hollow={false}
                  icon={<MdClose />}
                  inverted={false}
                  onClick={[Function]}
                  position="right"
                >
                  <Button__StyledButton
                    border={2}
                    className="c2 c3 c4"
                    fontSize={3}
                    hollow={false}
                    icon={<MdClose />}
                    inverted={false}
                    onClick={[Function]}
                    position="right"
                  >
                    <button
                      className="c2 c3 c4 c5"
                      fontSize={3}
                      icon={<MdClose />}
                      onClick={[Function]}
                    >
                      <MdClose>
                        <IconBase
                          viewBox="0 0 40 40"
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            preserveAspectRatio="xMidYMid meet"
                            style={
                              Object {
                                "color": undefined,
                                "verticalAlign": "middle",
                              }
                            }
                            viewBox="0 0 40 40"
                            width="1em"
                          >
                            <g>
                              <path
                                d="m31.6 10.7l-9.3 9.3 9.3 9.3-2.3 2.3-9.3-9.3-9.3 9.3-2.3-2.3 9.3-9.3-9.3-9.3 2.3-2.3 9.3 9.3 9.3-9.3z"
                              />
                            </g>
                          </svg>
                        </IconBase>
                      </MdClose>
                    </button>
                  </Button__StyledButton>
                </Button>
              </IconButton__StyledIconButton>
            </IconButton>
          </Modal__StyledCloseButton>
        </div>
      </Modal__StyledModalContent>
    </div>
  </Modal__StyledModal>
</Modal>
`;
